src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -2464,3 +2464,32 @@ impl Step for ReproducibleArtifacts {
         if added_anything { Some(tarball.generate()) } else { None }
     }
 }
+
+/// Tarball containing a prebuilt version of the libgccjit library,
+/// needed as a dependency for the GCC codegen backend (similarly to the LLVM
+/// backend needing a prebuilt libLLVM).
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Gcc {
+    pub target: TargetSelection,
+}
+
+impl Step for Gcc {
+    type Output = GeneratedTarball;
+
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("gcc")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(Gcc { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let tarball = Tarball::new(builder, "gcc", &self.target.triple);
+        let output = builder.ensure(super::gcc::Gcc { target: self.target });
+        tarball.add_file(output.libgccjit, ".", 0o644);
```

Comment:
How would this act from a tarball source where `gcc` sources missing?

Reply:
The GCC build would fail. Maybe we could try to detect this (directory missing/empty) and print a warning that GCC sources have to be provided?

Reply:
`gcc::Gcc`  does `builder.config.update_submodule("src/gcc");`. If we switch that to `require_submodule`, I thought that it would print an error if the submodule directory is missing, but in tarballs this call is just ignored.

So probably a manual warning it is.

Reply:
I think we should hard-error to avoid skipping it silently. But due to `const DEFAULT: bool = true;` it will run by default, which means that hard-error will cause `x dist` to fail for anyone running on tarball sources.

I think `x dist` shouldn't run this by default (similar to `x build`). If it's not a default dist step then hard-error on missing gcc sources should be fine.

Reply:
Good point. Is there a way to run "all default + something extra"? I would like to just run `x dist` on CI without having to think about running `x dist gcc` as an extra step. But if it's not easily possible, I'll just add `x dist gcc` explicitly to CI.

Or, as an alternative, maybe we could produce a warning if the sources are missing if GCC wasn't requested explicitly, bur an error when it was requested explicitly?

Reply:
Both sounds good to me. Personally, I would go with the "running `x dist gcc` as an extra step" on some runners if we don't want to build `dist::Gcc` step on every dist runner.

Reply:
Actually there is `--include-default-paths`, we already use it. So I will just add `gcc` to the x64 dist job and nowhere else. Thanks!